lib/mapping/MapFormatH3M.cpp

```diff
@@ -1256,7 +1256,7 @@ void CMapLoaderH3M::readObjects()
 				readMessageAndGuards(res->message, res, objPos);
 
 				res->amount = reader->readUInt32();
-				if(objTempl->subid == Res::GOLD)
+				if(objTempl->subid == vstd::to_underlying(ECurrencyID::GOLD))

```

Comment:
I would probably replace all to_underlying with getNum(), if only for consistency with existing code.
```suggestion
				if(objTempl->subid == CurrencyID(ECurrencyID::GOLD).getNum())
```

Other than this - good to go I think?

Reply:
@IvanSavenko I will, but why? I think using standard library function is good too (and std::to_underlying is part of C++23).

Reply:
From my point it is not about library vs custom (Identifier will use to_underlying internally anyway).
The way I see it, this code has constant that should not have been hardcoded.
Proper code that we (eventually) should have here is something like
```c++
    CurrencyID resourceType(objTempl->subID);
    if (resourceType->pileMultiplier != 1)
        amount = amount * resourceType->multiplier;
```
Meaning, this code should use CurrencyID, but its value should come from elsewhere instead of being hardcoded to be gold

Reply:
Or, in other words, I see enumeration values embedded in code a necessary evil at this point (but still huge improvement over hardcoded numeric constants we had originally)
Meaning, we shouldn't add code that is specific to enum's (such as to_underlying)

Reply:
Agree with you on anything but resources, because resources will not be able to load from mods AFAIK.

Reply:
Maybe. But I'd rather keep possibility for new resources open. There is no need to have mithril hardcoded & we eventually may want new resources, e.g. for something like H5-like necromancy.

And at the very least, I'd like to have handler class for resources, if only for better organization of resource-specific data (initial resources, market prices, icons, etc) instead of having it scattered across entire codebase.

Reply:
All to_underlying for ECurrencyID replaced.